ZJIT: Panic unimplemented for OOB basic block args

havenwood opened the pull request:
- Go from 8 to 10 `ALLOC_REGS` registers for ARM64 and x86_64
- Panic about unimplemented register spilling when block args are out of bounds

Testing ZJIT I noticed an index out of bounds panic with:

```sh
ruby --zjit -e "pp 0"
```

Pending the implementation of register spilling, I propose:
1) Improve the panic message when there aren't enough registers for block arguments
2) Increase the number of basic block registers from 8 to 10 to cover many more cases in the interim.

k0kubun commented on zjit/src/codegen.rs:
```diff
@@ -730,9 +730,26 @@ fn gen_push_frame(asm: &mut Assembler, recv: Opnd) {
+    use super::*;
+
+    #[test]
+    #[should_panic(expected = "register spilling not yet implemented, too many basic block arguments (11/10)")]
```
The panic (by `unimplemented!`) itself works as an inline test, so I don't think you should test the test. It's also tedious to update the "11/10" part whenever you touch the `ALLOC_REGS` constant.

havenwood commented:
Thank you for explaining why I was in error, @k0kubun. I very much appreciate your thoughtful response. I've updated the PR accordingly, removing the new registers and unnecessary test.

I updated this PR to only include the unimplemented panic message.

k0kubun reviewed (approved):
Thanks :+1:
